build.gradle

```diff
@@ -0,0 +1,137 @@
+    apply plugin: 'maven'
+    apply plugin: 'idea'
+    apply plugin: 'java'
+    apply plugin: "signing"
+    apply plugin: "jacoco"
+    // TODO(bdrutu): enable all checks.
+    apply plugin: "net.ltgt.errorprone"
+
+    group = "com.google.instrumentation"
+    version = "0.1.0-SNAPSHOT" // CURRENT_INSTRUMENTATION_VERSION
+
+    sourceCompatibility = 1.6
+    targetCompatibility = 1.6
+
+    repositories {
+        mavenCentral()
+        mavenLocal()
+    }
+
+    [compileJava, compileTestJava].each() {
+        it.options.compilerArgs += ["-Xlint:all", "-Xlint:-try", "-Xlint:-path"]
+        it.options.encoding = "UTF-8"
+        if (rootProject.hasProperty('failOnWarnings') && rootProject.failOnWarnings.toBoolean()) {
+            it.options.compilerArgs += ["-Werror"]
+        }
+    }
+
+    compileTestJava {
+        // serialVersionUID is basically guaranteed to be useless in tests
+        options.compilerArgs += ["-Xlint:-serial"]
+        // It undeprecates DoubleSubject.isEqualTo(Double).
+        options.compilerArgs += ["-Xlint:-deprecation"]
+        sourceCompatibility = 1.7
+        targetCompatibility = 1.7
+    }
+
+    jar.manifest {
+        attributes('Implementation-Title': name,
+                'Implementation-Version': version,
+                'Built-By': System.getProperty('user.name'),
+                'Built-JDK': System.getProperty('java.version'),
+                'Source-Compatibility': sourceCompatibility,
+                'Target-Compatibility': targetCompatibility)
+    }
+
+    javadoc.options {
+        encoding = 'UTF-8'
+        links 'https://docs.oracle.com/javase/8/docs/api/'
+    }
+
+    ext {
+        guavaVersion = '19.0'
+        protobufVersion = '3.2.0'
+        grpcContextVersion = '1.1.2'
+
+        libraries = [
+                errorprone   : 'com.google.errorprone:error_prone_annotations:2.0.11',
+                grpc_context : "io.grpc:grpc-context:${grpcContextVersion}",
+                guava        : "com.google.guava:guava:${guavaVersion}",
+                jsr305       : 'com.google.code.findbugs:jsr305:3.0.0',
+                protobuf     : "com.google.protobuf:protobuf-java:${protobufVersion}",
+
+                // Test dependencies.
+                guava_testlib: "com.google.guava:guava-testlib:${guavaVersion}",
+                junit        : 'junit:junit:4.11',
+                mockito      : 'org.mockito:mockito-core:1.9.5',
+                truth        : 'com.google.truth:truth:0.30',
+        ]
+    }
+
+    dependencies {
+        testCompile libraries.guava_testlib,
+                libraries.junit,
+                libraries.mockito,
+                libraries.truth
+
+        // The ErrorProne plugin defaults to the latest, which would break our
+        // build if error prone releases a new version with a new check
+        errorprone 'com.google.errorprone:error_prone_core:2.0.15'
+    }
+
+    checkstyle {
+        configFile = file("$rootDir/checkstyle.xml")
```

Comment:
No. It is a bit hard to configure severity level and ignoring things like SingleJavadocLine. gRPC did the same thing copying the checkstyle from the repository. So this should be fine. @zhangkun83 is this the same thing you did correct?